components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/InMemoryAPIDeployer.java

```diff
@@ -72,15 +73,24 @@ public boolean deployAPI(String apiId, String gatewayLabel) {
                     if (gatewayRuntimeArtifact != null) {
                         GatewayAPIDTO gatewayAPIDTO = new Gson().fromJson(gatewayRuntimeArtifact, GatewayAPIDTO.class);
                         apiGatewayAdmin.deployAPI(gatewayAPIDTO);
+                        if (debugEnabled) {
+                            log.debug("API with " + apiId + " is deployed in gateway with the label of " + gatewayLabel);
+                        }
                         return true;
                     } else {
-                        log.error("Error retrieving artifacts for API " + apiId + ". Storage returned null");
+                        String msg = "Error retrieving artifacts for API " + apiId + ". Storage returned null";
+                        log.error(msg);
+                        throw new ArtifactSynchronizerException(msg);
                     }
                 } catch (IOException | ArtifactSynchronizerException e) {
-                    log.error("Error deploying " + apiId + " in Gateway", e);
+                    String msg = "\"Error deploying \" + apiId + \" in Gateway\"";
```

Comment:
Do we need these escapes?

Reply:
Fixed https://github.com/wso2/carbon-apimgt/pull/9019/commits/ae94c47533b0f41ccd7818f935d646fa9926bb0d